src/axom/inlet/InletVector.hpp

```diff
@@ -0,0 +1,124 @@
+ *******************************************************************************
+ */
+struct InletVector
+{
+  primal::Vector3D vec;
+  int dim = 3;
+
+  /*!
+   *******************************************************************************
+   * \brief Constructs an empty vector (size defaults to 3)
+   *******************************************************************************
+   */
+  InletVector() = default;
+
+  /*!
+   *******************************************************************************
+   * \brief Constructs a vector with an initializer list
+   * 
+   * \param [in] values The vector components to construct with
+   *******************************************************************************
+   */
+  InletVector(std::initializer_list<double> values)
+    : vec(values)
+    , dim(values.size())
+  { }
+
+  /*!
+   *******************************************************************************
+   * \brief Constructs a vector with an existing Primal vector and a dimension
+   * 
+   * \param [in] v The existing Primal vector
+   * \param [in] d The dimension of the vector
+   *******************************************************************************
+   */
+  InletVector(primal::Vector3D&& v, int d = 3) : vec(std::move(v)), dim(d) { }
+
+  /*!
+   *******************************************************************************
+   * \brief Retrieves an element of the vector
+   * 
+   * \param [in] i The index of the element to retrieve (zero-indexed)
+   *******************************************************************************
+   */
+  double operator[](const int i) const { return vec[i]; }
+  /// \overload
+  double& operator[](const int i) { return vec[i]; }
+
+  /*!
+   *******************************************************************************
+   * \brief Retrieves the underlying Primal vector
+   *******************************************************************************
+   */
+  operator axom::primal::Vector3D &() { return vec; }
+  /// \overload
+  operator const axom::primal::Vector3D &() const { return vec; }
+};
+
+/*!
+ *******************************************************************************
+ * \brief Compares two InletVectors
+ * 
+ * \param [in] u The comparison LHS
+ * \param [in] v The comparison RHS
+ * 
+ * \return Whether the two vectors are equivalent
+ *******************************************************************************
+ */
+inline bool operator==(const InletVector& u, const InletVector& v)
+{
+  return (u.vec == v.vec) && (u.dim == v.dim);
+}
+
+/*!
+ *******************************************************************************
+ * \brief "Prints" the vector to a stream
+ * 
+ * \param [inout] os The stream to insert into
+ * \param [in] v The vector to insert into the stream
+ *******************************************************************************
+ */
+inline std::ostream& operator<<(std::ostream& os, const InletVector& v)
+{
+  os << v.vec;
```

Comment:
Should this function deal with `v.dim` ?  E.g. possibly printing the dimension and/or only printing the first `dim` values?

Reply:
I think printing the first `dim` values is reasonable